### PR TITLE
Issue 2318 -  Functionality to delete grader

### PR DESCRIPTION
**Pull request by jeremygoh**
Functionality to delete a grader for #2318

### CODE REVIEW

File: app/controllers/tas_controller.rb [hunk 1]
```diff
@@ -17,6 +17,17 @@ def edit
     @user = Ta.find_by_id(params[:id])
   end
 
+  def destroy
+    @user = Ta.find_by_id(params[:id])
```
Comment (david-yz-liu) [inline app/controllers/tas_controller.rb]: You should be able to use `find` here

File: app/views/tas/_tas_table.js.jsx.erb [hunk 2]
```diff
@@ -65,6 +71,7 @@
         t['user_name'] = ta.user_name;
         t['first_name'] = ta.first_name;
         t['edit_link'] = <a href={ta.edit_link}><%= j raw I18n.t(:edit) %></a>
+        t['delete_link'] = <a href={ta.delete_link} data-method="delete" rel="nofollow" data-confirm="Are you sure?"><%= j raw I18n.t(:delete) %></a>
```
Comment (david-yz-liu) [inline app/views/tas/_tas_table.js.jsx.erb]: The "Are you sure" should be internationalized.

File: app/controllers/tas_controller.rb [hunk 3]
```diff
@@ -17,6 +17,17 @@ def edit
...
+    if @user.destroy
```
Comment (david-yz-liu) [inline app/controllers/tas_controller.rb]: I think this is a runtime error if the TA isn't found - you can flash an error in that case.

**Comment by david-yz-liu** [comment]
Thanks, @jeremygoh. The other thing which is missing is a call to renegerate the repo permissions - if a TA is deleted, they should lose access to any repos they had been given access to previously.

**Comment by jeremygoh** [comment]
Thanks for the quick review @david-yz-liu !

I've made the changes to the `destroy` method in the `tas_controller`, and also internationalized the confirmation dialog as requested (I knew I was forgetting to internationalize somewhere!).

For the regeneration of the repo permissions, is this call sufficient?
https://github.com/jeremygoh/Markus/blob/issue-2318-delete-grader/app/models/ta.rb#L10
https://github.com/jeremygoh/Markus/blob/issue-2318-delete-grader/app/models/user.rb#L299-L306

**Comment by david-yz-liu** [comment]
Thanks, @jeremygoh!

**Pull request merged**